### PR TITLE
Add dism, dvmp to discovery service and tree view

### PR DESCRIPTION
**Pull request by djwhatle**
Quite a few lines added, but this is just a direct copy-paste of code already in the discovery service.

image

New endpoints added. Tree view now shows the complete view of MTC CRs,

**DVMP**
```
/namespaces/openshift-migration/directvolumemigrationprogresses
/namespaces/openshift-migration/directvolumemigrationprogress/my-dvmp-name (currently 404ing)
```

**DISM**
```
/namespaces/openshift-migration/directimagestreammigrations
/namespaces/openshift-migration/directimagestreammigration/my-dism-name (currently 404ing)
```

**Tree view with DVMP and DISM**
```bash
AUTH_OPTIONAL=1 ROLE=discovery make run-fast
curl localhost:8080/namespaces/openshift-migration/plans/plan-rocket-chat/tree | jq
```

```json
    {
      "kind": "Migration",
      "namespace": "openshift-migration",
      "name": "dc1b8da0-931d-11eb-a725-0f1f65d7ed65",
      "children": [
        {
          "kind": "DirectVolumeMigration",
          "namespace": "openshift-migration",
          "name": "dc1b8da0-931d-11eb-a725-0f1f65d7ed65-q4q8q",
          "children": [
            {
              "kind": "DirectVolumeMigrationProgress",
              "namespace": "openshift-migration",
              "name": "49e187c6e58b404f60caf687d18cb2bf",
              "objectLink": "/namespaces/openshift-migration/directvolumemigrationprogresses/49e187c6e58b404f60caf687d18cb2bf"
            }
          ],
          "objectLink": "/namespaces/openshift-migration/directvolumemigrations/dc1b8da0-931d-11eb-a725-0f1f65d7ed65-q4q8q"
        },
        {
          "kind": "DirectImageMigration",
          "namespace": "openshift-migration",
          "name": "dc1b8da0-931d-11eb-a725-0f1f65d7ed65-7xz42",
          "objectLink": "/namespaces/openshift-migration/directimagemigrations/dc1b8da0-931d-11eb-a725-0f1f65d7ed65-7xz42"
        }
      ],
      "objectLink": "/namespaces/openshift-migration/migrations/dc1b8da0-931d-11eb-a725-0f1f65d7ed65"
    }

```

### CODE REVIEW

File: pkg/controller/discovery/web/direct.go [hunk 1]
```diff
@@ -123,6 +164,76 @@ func (h *DirectImageMigrationHandler) Prepare(ctx *gin.Context) int {
...
+//
+// Prepare to fulfil the request.
+// Fetch the referenced dvmp.
+// Perform SAR authorization.
+func (h *DirectVolumeMigrationProgressHandler) Prepare(ctx *gin.Context) int {
+	status := h.BaseHandler.Prepare(ctx)
+	if status != http.StatusOK {
+		return status
+	}
+	name := ctx.Param(DirectVolumeParam)
```
Comment (jortel) [inline pkg/controller/discovery/web/direct.go:211]: This needs to be _DirectVolumeProgressParam_.

Comment (djwhatle) [reply pkg/controller/discovery/web/direct.go]: Thanks Jeff. Good catch.

**Comment by djwhatle** [comment]
https://user-images.githubusercontent.com/7576968/113610010-9222c380-961a-11eb-9d28-e619acc45d14.mov

**Review by jortel — APPROVED** [review]
LGTM

**Pull request merged**